go/vt/vtgate/resilient_srv_topo_server.go

```diff
@@ -280,33 +302,64 @@ func (server *ResilientSrvTopoServer) GetSrvKeyspace(ctx context.Context, cell,
 	entry.mutex.Lock()
 	defer entry.mutex.Unlock()
 
-	// If the entry is fresh enough, return it
-	if time.Now().Sub(entry.insertionTime) < server.cacheTTL {
+	// If the entry exists, return it
+	if !entry.insertionTime.IsZero() {
 		return entry.value, entry.lastError
 	}
 
-	// not in cache or too old, get the real value
+	// not in cache, get the real value
 	newCtx, cancel := context.WithTimeout(context.Background(), *srvTopoTimeout)
 	defer cancel()
 
-	result, err := server.topoServer.GetSrvKeyspace(newCtx, cell, keyspace)
+	// start watching
+	notifications, _, err := server.topoServer.WatchSrvKeyspace(newCtx, cell, keyspace)
 	if err != nil {
+		// set error if there is no cached value
 		if entry.insertionTime.IsZero() {
```

Comment:
Change to test entry.value == nil.
The logic here is we should still return cached value (if any) when topo server is down.